### PR TITLE
Bug: Incorrect plugin insertion order in settings.gradle with pluginManagement and buildscript

### PR DESCRIPTION
**Pull request by div-bargali**
## What's changed?

Added failing unit tests to demonstrate incorrect plugin ordering in `settings.gradle[.kts]` files when both `pluginManagement` and `buildscript` blocks are present.

**New tests added:**
- `addPluginWithPluginManagementAndBuildscriptBlocks()` - Groovy DSL test
- `addPluginWithPluginManagementAndBuildscriptBlocksKotlin()` - Kotlin DSL test

These tests currently fail because `AddPluginVisitor` incorrectly inserts the `plugins` block between `pluginManagement` and `buildscript`, violating Gradle's required ordering.

## What's your motivation?

- Closes https://github.com/openrewrite/rewrite/issues/6479

When `AddSettingsPlugin` is applied to a `settings.gradle` file containing both `pluginManagement` and `buildscript` blocks, it produces invalid Gradle syntax that causes build failures with the error:
only buildscript {} and other plugins {} script blocks are allowed before plugins {} block

## Anything in particular you'd like reviewers to focus on?

- Verify that the test cases accurately represent the expected Gradle ordering requirements
- Confirm that both Groovy and Kotlin DSL tests properly cover the edge case
- Review the `interpolateResolvedVersionKotlin()` helper method for Kotlin tests

## Any additional context

This PR includes the failing tests to document the bug before implementing the fix. The fix will be submitted in a follow-up commit.

**Impact:** This issue affects any project using `AddSettingsPlugin` on settings files that contain both `pluginManagement` and `buildscript` blocks.

### Checklist
- [x] I've added unit tests to cover both positive and negative cases
- [x] I've read and applied the [recipe conventions and best practices](https://docs.openrewrite.org/authoring-recipes/recipe-conventions-and-best-practices)
- [x] I've used the IntelliJ IDEA auto-formatter on affected files

### CODE REVIEW

**Comment by timtebeek** [comment]
Thanks for getting this started @div-bargali ; At first glance these tests look good to me!

File: rewrite-gradle/src/test/java/org/openrewrite/gradle/plugins/AddSettingsPluginTest.java [hunk 1]
```diff
@@ -154,4 +246,13 @@ private static Consumer<SourceSpec<G.CompilationUnit>> interpolateResolvedVersio
...
         });
     }
+
+    private static Consumer<SourceSpec<K.CompilationUnit>> interpolateResolvedVersionKotlin(@Language("kotlin") String after) {
+        return spec -> spec.after(actual -> {
+            assertThat(actual).isNotNull();
+            Matcher version = Pattern.compile("3\\.\\d+(\\.\\d+)?").matcher(actual);
+            assertThat(version.find()).isTrue();
+            return after.formatted(version.group(0));
+        });
```
Comment (timtebeek) [inline rewrite-gradle/src/test/java/org/openrewrite/gradle/plugins/AddSettingsPluginTest.java]: Elsewhere we've adopted the pattern of using the new-ish `.actual()` method on AssertJ; perhaps that's helpful here too.
```java
return assertThat(actual)
  .containsPattern("3\\.\\d+(\\.\\d+)?")
  .actual();
```

**Comment by shanman190** [comment]
Both tests also look good to me too!

**Comment by div-bargali** [comment]
Added the fix for the bug @timtebeek.

I also noticed that in the Kotlin script handling section, the AST was parsed incorrectly for the (.kts) files. Due to this, the kotlin login in `AddPluginVistor` was not working correctly for `build.gradle.kts` file. I was able to assert that by adding a simple-but-missing test in `AddBuildPluginTest` class.

I have included the fix for that too in this PR.

File: rewrite-gradle/src/test/java/org/openrewrite/gradle/plugins/AddSettingsPluginTest.java (hunk 1 above)
Comment (div-bargali) [reply rewrite-gradle/src/test/java/org/openrewrite/gradle/plugins/AddSettingsPluginTest.java]: Added this pattern in the test.

File: rewrite-gradle/src/test/java/org/openrewrite/gradle/plugins/AddSettingsPluginTest.java [hunk 2]
```diff
@@ -146,11 +148,114 @@ void addPluginApplyFalse() {
...
+    }
+
+    private static Consumer<SourceSpec<K.CompilationUnit>> interpolateResolvedVersionKotlin(@Language("kotlin") String after) {
+        return spec -> spec.after(actual -> {
+            assertThat(actual)
+                .isNotNull()
+                .containsPattern("3\\.\\d+(\\.\\d+)?");
             Matcher version = Pattern.compile("3\\.\\d+(\\.\\d+)?").matcher(actual);
-            assertThat(version.find()).isTrue();
+            version.find();
```
Comment (timtebeek) [inline rewrite-gradle/src/test/java/org/openrewrite/gradle/plugins/AddSettingsPluginTest.java]: Let's retain the original check on the return value of `find()`.
```suggestion
            assertThat(version.find()).isTrue();
```

File: rewrite-gradle/src/test/java/org/openrewrite/gradle/plugins/AddSettingsPluginTest.java [hunk 3]
```diff
@@ -146,11 +148,114 @@ void addPluginApplyFalse() {
...
+    }
+
     private static Consumer<SourceSpec<G.CompilationUnit>> interpolateResolvedVersion(@Language("groovy") String after) {
         return spec -> spec.after(actual -> {
-            assertThat(actual).isNotNull();
+            assertThat(actual)
+                    .isNotNull()
+                    .containsPattern("3\\.\\d+(\\.\\d+)?");
+            Matcher version = Pattern.compile("3\\.\\d+(\\.\\d+)?").matcher(actual);
+            version.find();
```
Comment (timtebeek) [inline rewrite-gradle/src/test/java/org/openrewrite/gradle/plugins/AddSettingsPluginTest.java]: Let's retain the original check on the return value of `find()`.
```suggestion
            assertThat(version.find()).isTrue();
```

**Review by timtebeek — APPROVED** [review]
Thanks a lot for the help here @div-bargali ; that a parallel effort was started accidentally only speaks to that you've picked a worthwhile issue to fix, and I like the approach you took here.
